examples/Land/soilmodel.jl

```diff
@@ -0,0 +1,89 @@
+using StaticArrays
+using CLIMA.VariableTemplates
+import CLIMA.DGmethods: BalanceLaw,
+                        vars_aux, vars_state, vars_gradient, vars_diffusive,
+                        flux_nondiffusive!, flux_diffusive!, source!,
+                        gradvariables!, diffusive!,
+                        init_aux!, init_state!,
+                        boundary_state!, wavespeed, LocalGeometry
+using CLIMA.DGmethods.NumericalFluxes: NumericalFluxNonDiffusive,
+                                       NumericalFluxDiffusive,
+                                       GradNumericalPenalty
+
+"""
+    SoilModel
+
+Computes diffusive flux `F` in:
+
+```math
+∂(ρcT)   ∂      ∂T
+------ = --(λ * --)
+  ∂t     ∂z     ∂z
+```
+where
+
+ - `ρ` is the density of the soil (kg/m³)
+ - `c` is the soil heat capacity (J/(kg K))
+ - `λ` is the thermal conductivity (W/(m K))
+"""
+struct SoilModel <: BalanceLaw
+  ρ::Float64
+  c::Float64
+  λ::Float64
+  surfaceT::Float64
+  initialT::Float64
+end
+
+# Stored in the aux state are:
+#   `coord` coordinate points (needed for BCs)
+#   `u` advection velocity
+#   `D` Diffusion tensor
+vars_aux(::SoilModel, FT) = @vars(z::FT, ρ::FT, c::FT, λ::FT) # stored dg.auxstate
+vars_state(::SoilModel, FT) = @vars(ρcT::FT) # stored in Q
+vars_gradient(::SoilModel, FT) = @vars(T::FT) # not stored
+vars_diffusive(::SoilModel, FT) = @vars(∂T∂z::FT) # stored in dg.diffstate
+
+function gradvariables!(m::SoilModel, transform::Vars, state::Vars, aux::Vars, t::Real)
+  transform.T = state.ρcT / (m.ρ * m.c)
+end
+function diffusive!(m::SoilModel, diffusive::Vars, ∇transform::Grad, state::Vars, aux::Vars, t::Real)
+  diffusive.∂T∂z = ∇transform.T[2]
+end
+function flux_nondiffusive!(m::SoilModel, flux::Grad, state::Vars, aux::Vars, t::Real)
+end
+function flux_diffusive!(m::SoilModel, flux::Grad, state::Vars, diffusive::Vars, aux::Vars, t::Real)
+  flux.ρcT += SVector(0, m.λ * diffusive.∂T∂z, 0)
```

Comment:
Maybe better to make this coordinate independent, using the `vertical_unit_vector` from `orientation.jl`, i.e., 
```suggestion
  flux.ρcT += m.λ * diffusive.∂T∂z .* vertical_unit_vector
```
(To make sure this works on the sphere in the end.)